### PR TITLE
puppet-staging -> puppet-archive

### PR DESCRIPTION
**Pull request by angrox**
Migrated the module from puppet-staging to puppet-archive (the one oracle_java uses too). Secondary I added the possibility to define a checksum and checksum_type to test the downloaded archive file against.

### CODE REVIEW

**Comment by antoineco** [comment]
Don't forget extras.pp, it also uses staging. Actually in v0.4 I switched back from archive to staging because ~~archive didn't handle jar files, if I remember well.~~

Edit: actually that was not the reason, but I can't even remember why I did that. It had something to do with the formats Archive supported back then.
You can find my changes [here](https://github.com/antoineco/aco-tomcat/commit/5c4ab7a885e28831b68adc3ceb6edeb1ee0fb93d#diff-27faa283b1a26b458d1c8dd429985f73L9).

**Comment by angrox** [comment]
Ok, extra.pp needs the change, too. I will test with jar files., haven't done that
Edit: I have seen you already change extra.pp. I will test the changes and the deployment of jar files (tomorrow I have time for that)

**Comment by antoineco** [comment]
The changes are actually from long time ago, you basically have to revert them :)

**Comment by angrox** [comment]
Let's put this on hold for a few days. Looks like the puppet-archive module moved to the puppet-community. I am currently asking the developer if the module will be republished on the forge too.

**Comment by antoineco** [comment]
It's here https://forge.puppetlabs.com/puppet/archive

**Comment by angrox** [comment]
No. The new Repository is here: https://github.com/puppet-community/puppet-archive
That is the problem: It is completly unclear to me if the puppet-community (https://puppet.community/) is a community driven fork from the puppet forge. Hopefully they are also pushing their modules back to the forge and change the project url. If not it will leave a lot of forge modules with unsatisfied dependencies.

**Comment by antoineco** [comment]
Yes it's the same, it was forked and published under a different name, that's all.

**Comment by antoineco** [comment]
Hi @angrox, looks like I didn't get any notification for your new commits, sorry! Is that ready for review now?

**Comment by angrox** [comment]
Hi antoineco, 

sorry, I will continue my work on this. I will update my codebase and then try to bring the new patch up - please give me a day or two.

**Comment by antoineco** [comment]
No rush, just mention my name to catch my attention if needed :)

**Comment by angrox** [comment]
I will do. Thanks for your patience!

**Comment by angrox** [comment]
@antoineco Am I correct that the rspec tests via travis-ci are always failing?

**Comment by antoineco** [comment]
I think it's not even running because of conflicts with the master branch. You could try to rebase your branch and fix conflicts manually, maybe that would help.

**Comment by angrox** [comment]
@antoineco, I am ready that you have a look at it. I have changed the following: 
- Moved from puppet/staging to puppet/archive
- Fixed spec_helper.rb so local rspec tests will succeed (manifest and module directory)
- Added puppetversion a fact in spec_helper.rb for in puppet 4.3.x the fact is not prefilled when using puppet-rspec
- Changed the test against Fedora in archive_spec.rb from F21 to F23 with tomcat8
- Merged the latest commits

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1 +1,9 @@
 require 'puppetlabs_spec_helper/module_spec_helper'
+require 'rspec-puppet'
```
Comment (antoineco) [inline spec/spec_helper.rb]: This is implicitly included by the spec_helper above, probably also the whole block below (not sure about that)

Comment (angrox) [inline spec/spec_helper.rb]: I just tested it. You are right that the "require rspec-puppet" is obsolete (it is included in the puppetlabs spec helper). 
Nevertheless the rake task does not succeed if the directories for the fixtures are not defined (at least in my local environment).

File: metadata.json [hunk 2]
```diff
@@ -21,8 +21,8 @@
       "version_requirement": ">= 1.1.3 < 2.0.0"
     },
     {
-      "name": "nanliu/staging",
-      "version_requirement": ">= 1.0.0 < 2.0.0"
+      "name": "puppet-community/archive",
```
Comment (antoineco) [inline metadata.json]: Shouldn't it be `puppet/archive`?

File: spec/spec_helper.rb (hunk 1 above)
Comment (antoineco) [inline spec/spec_helper.rb]: Strange thing, it works for me with both Puppet 4.3 and 3.8 (fresh system running in Docker). I think you have some other Rspec settings interfering. Are you running the test with `bundle exec rake spec`?

File: metadata.json (hunk 2 above)
Comment (angrox) [inline metadata.json]: Yes, you are right. That changed from the last time I edited it. All puppet-community forge modules were renamed.

File: spec/spec_helper.rb (hunk 1 above)
Comment (angrox) [inline spec/spec_helper.rb]: Yes "bundle exec rake spec". Tested in my rvm environment with ruby 1.9.3 and 2.2.4

**Comment by angrox** [comment]
Heya, I just wanted to ask how we want to proceed? I am a little bit confused right now :-|

**Comment by antoineco** [comment]
Hey @angrox, could you reset `spec_helper.rb` and squash all the commits into one? After that I can review one more time and merge it :+1:

**Comment by angrox** [comment]
Done, as good as I could!

**Comment by antoineco** [comment]
Hm, I still see 10 commits, have you done a `git rebase -i HEAD~10`? From there you should replace all the `pick` by `s` or `squash`, but the first one. Example:

```
pick 916c7cb commit message
s a8627e2 commit messages
s f9ae49d commit message
s 5be5d1e commit message
[...]
s 6fe053a commit message
s 9371cd8 commit message
```

**Comment by angrox** [comment]
Sorry for the delay. I have rebased everything you mentioned. Could you please check it again? Thanks!

**Comment by antoineco** [comment]
Hey @angrox, I still see 14 commits unfortunately, did you push your changes?

**Comment by angrox** [comment]
Strange. As soon as I clone the repository I see only one commit from me where everything is in.

**Comment by angrox** [comment]
Ok. It's getting to complex. Let's close this pull request. I have a complete patch of my change, up to the current commit state of your master branch. I will open a new pull request where we have one clear commit.

**Pull request closed without merging**